nixos/services/haproxy/default.nix

```diff
@@ -6,9 +6,99 @@ let
   cfg = config.flyingcircus.services.haproxy;
   fclib = config.fclib;
 
+  # last returns the last element of a list or null if the list is empty
```

Comment:
please move the generic functions to nixos/lib in our overlay and access them through fclib.